rules/microsoft_rules/microsoft365_external_sharing.py

```diff
@@ -0,0 +1,44 @@
+import re
+from fnmatch import fnmatch
+
+from panther_base_helpers import m365_alert_context
+
+email_regex = re.compile(r"([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+")
+
+ALLOWED_DOMAINS = ["mycompany.com", "alloweddomain.com"]
+
+ALLOWED_USERS = ["exception@outsider.com"]
+
+ALLOWED_PATHS = ["*/External/*", "External/*"]
+
+
+def allowed_path(relative_url):
+    for path in ALLOWED_PATHS:
+        if fnmatch(relative_url, path):
+            return True
+    return False
+
+
+def rule(event):
+    if event.get("Operation", "") == "AnonymousLinkCreated":
+        return not allowed_path(event.get("SourceRelativeUrl"))
+    if event.get("Operation", "") == "AddedToSecureLink":
+        if allowed_path(event.get("SourceRelativeUrl")):
```

Comment:
lines 26 and 27 seem to be logically equivalent to line 24. It might make sense to re-use the spelling of this from line 24 in place of current lines 26-27.

Reply:
If not allowed_path, i would need to perform further checks so cannot use same as 24